src/dotnet/APIView/APIViewWeb/DTOs/SiteNotificationDto.cs

```diff
@@ -1,44 +1,91 @@
-using System;
-using APIViewWeb.Models;
+using System.Text.Json.Serialization;
 
-namespace APIViewWeb.DTOs;
-
-public class SiteNotificationDto
+namespace APIViewWeb.DTOs
 {
-    public string ReviewId { get; set; }
-    public string RevisionId { get; set; }
-    public string Title { get; set; }
-    public string Summary { get; set; }
-    public string Message { get; set; }
-
-    private string _status;
-    public string Status
+    public class SiteNotificationDto
+    {
+        [JsonPropertyName("reviewId")]
+        public string ReviewId { get; set; }
+
+        [JsonPropertyName("revisionId")]
+        public string RevisionId { get; set; }
+
+        [JsonPropertyName("summary")]
+        public string Summary { get; set; }
+
+        [JsonPropertyName("message")]
+        public string Message { get; set; }
+
+        [JsonPropertyName("type")]
+        [JsonConverter(typeof(JsonStringEnumConverter))]
+        public SiteNotificationType Type { get; set; }
+
+        [JsonPropertyName("toastNotification")]
+        public ToastNotificationDto ToastNotification { get; set; }
+
+
+        private string _status;
+
+        [JsonPropertyName("status")]
+        public string Status
+        {
+            get => _status;
+            set => _status = NormalizeStatus(value);
+        }
+
+        private static string NormalizeStatus(string value)
+        {
+            if (string.IsNullOrWhiteSpace(value))
+            {
+                return string.Empty;
+            }
+
+            string normalized = value.Trim().ToLowerInvariant();
+
+            // Return known values as constants, allow unknown values to pass through
+            return normalized switch
+            {
+                "success" => SiteNotificationStatus.Success,
+                "info" => SiteNotificationStatus.Info,
+                "error" => SiteNotificationStatus.Error,
+                "warning" => SiteNotificationStatus.Warning,
+                _ => value.Trim() // Keep original casing for unknown values
+            };
+        }
+    }
+
+    public class ToastNotificationDto
     {
-        get => _status;
-        set => _status = NormalizeStatus(value);
+        [JsonPropertyName("title")]
+        public string Title { get; set; }
+        [JsonPropertyName("message")]
+        public string Message { get; set; }
+        [JsonConverter(typeof(JsonStringEnumConverter))]
+        [JsonPropertyName("action")]
+        public SiteNotificationAction Action { get; set; }
```

Comment:
I think you should also decide this on client side, no need to send it from the server

Reply:
and what are the parameters to take this decision on the client side? knowing the type of notification?